pkg/cmd/readyz/readyz.go

```diff
@@ -0,0 +1,189 @@
+func NewReadyzCommand() *cobra.Command {
+	opts := readyzOpts{
+		listenPort: uint16(defaultListenPort),
+	}
+	cmd := &cobra.Command{
+		Use:   "readyz",
+		Short: "Serve the HTTP /readyz endpoint health check for an etcd member",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := opts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	opts.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (r *readyzOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Uint16Var(&r.listenPort, "listen-port", r.listenPort, "Listen on this port. Default 9980")
+	fs.DurationVar(&r.dialTimeout, "dial-timeout", defaultHTTPDialTimeout, "Dial timeout for the client. Default 2s")
+	fs.StringVar(&r.targetEndpoint, "target", defaultEndpoint, "Target endpoint to perform health check against. Default https://localhost:2379")
+	fs.StringVar(&r.clientCertFile, "cert-file", "", "Health probe TLS client certificate file. (required)")
+	fs.StringVar(&r.clientKeyFile, "key-file", "", "Health probe TLS client key file. (required)")
+	fs.StringVar(&r.clientCACertFile, "cacert-file", "", "Health probe TLS client CA certificate file. (required)")
+}
+
+// Validate verifies the inputs.
+func (r *readyzOpts) Validate() error {
+	if len(r.targetEndpoint) == 0 {
+		return fmt.Errorf("missing required flag: --target")
+	}
+	if len(r.clientKeyFile) == 0 {
+		return errors.New("missing required flag: --key-file")
+	}
+	if len(r.clientCertFile) == 0 {
+		return errors.New("missing required flag: --cert-file")
+	}
+	if len(r.clientCACertFile) == 0 {
+		return errors.New("missing required flag: --cacert-file")
+	}
+
+	return nil
+}
+
+// Run contains the logic of the readyz command which checks the health of the etcd member
+func (r *readyzOpts) Run() error {
+	shutdownCtx, cancel := context.WithCancel(context.Background())
+	shutdownHandler := server.SetupSignalHandler()
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/readyz", r.getReadyzHandlerFunc(shutdownCtx))
+	// Handle the /healthz endpoint as well since the static pod controller's guard pods check the /healthz endpoint
+	// https://github.com/openshift/library-go/blob/edab248e63516c65a93467eaa8224c86d69f5de9/pkg/operator/staticpod/controller/guard/manifests/guard-pod.yaml#L44
+	mux.HandleFunc("/healthz", r.getReadyzHandlerFunc(shutdownCtx))
+
+	addr := fmt.Sprintf("0.0.0.0:%d", r.listenPort)
+	klog.Infof("Listening on %s", addr)
+
+	server := &http.Server{
+		Addr:        addr,
+		Handler:     mux,
+		BaseContext: func(_ net.Listener) context.Context { return shutdownCtx },
+	}
+	go func() {
+		defer cancel()
+		<-shutdownHandler
+		klog.Infof("Received SIGTERM or SIGINT signal, shutting down server.")
+		server.Shutdown(shutdownCtx)
+	}()
+
+	c := net.ListenConfig{}
+	c.Control = permitAddressReuse
+	ln, err := c.Listen(shutdownCtx, "tcp", addr)
+	if err != nil {
+		return err
+	}
+	err = server.ServeTLS(ln, r.clientCertFile, r.clientKeyFile)
```

Comment:
`ServeTLS` takes a serving cert/key pair, not the client cert/key pair.  I don't mind if you re-use the existing etcd serving cert/key pair